src/ccm/ucc_ccm.h

```diff
@@ -0,0 +1,103 @@
+/*
+* Copyright (C) Mellanox Technologies Ltd. 2001-2020.  ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#ifndef UCC_CCM_H_
+#define UCC_CCM_H_
+
+#include <ucs/config/types.h>
+#include <ucs/debug/log_def.h>
+#include <ucs/config/parser.h>
+#include <assert.h>
+#include <string.h>
+#include "config.h"
+#include "api/ucc.h"
+#include "core/ucc_lib.h"
+#include "utils/ucc_component.h"
+
+typedef struct ucc_ccm_lib     ucc_ccm_lib_t;
+typedef struct ucc_ccm_iface   ucc_ccm_iface_t;
+typedef struct ucc_ccm_context ucc_ccm_context_t;
+typedef struct ucc_ccm_team    ucc_ccm_team_t;
+
+typedef struct ucc_ccm_lib_config {
+    /* Log level above which log messages will be printed */
+    ucs_log_component_config_t log_component;
+    /* Team library priority */
+    int                        priority;
+} ucc_ccm_lib_config_t;
+extern ucs_config_field_t ucc_ccm_lib_config_table[];
+
+typedef struct ucc_ccm_context_config {
+    ucc_ccm_iface_t *iface;
+    ucc_ccm_lib_t   *ccm_lib;
+} ucc_ccm_context_config_t;
+extern ucs_config_field_t ucc_ccm_context_config_table[];
+
+typedef struct ucc_ccm_iface {
+    ucc_component_iface_t          super;
+    char*                          name;
+    int                            priority;
+    ucc_lib_params_t               params;
```

Comment:
probably needs to be a pointer... if for no other reason - at least for ABI compatibility (if params grow)